### DISCUSSION TITLE
Locks Entity State and Service

### DISCUSSION THREAD

**Discussion by vmcosco** [G-Two/homeassistant-subaru#43] [General]
Everything is working great wit this integration, especially the remote start, but the locks are causing me issues.

This is on a 2021 Outback.  The state of the `lock.subaru_outback_door_locks` entity is showing as "unknown". 

I tried remote lock and unlock using more-info (as in your lovelace.yaml) but no luck.  I also tried the following tap action, with no luck:

```
        tap_action:
          action: call-service
          service: lock.lock
          target:
            entity_id: lock.subaru_outback_door_locks
```

Below is the error that I am getting.  I did change my device name from VictorC Outback to Subaru Outback (and renamed all entities).  Not sure if that could be causing any issues.

```
Logger: homeassistant.core
Source: custom_components/subaru/remote_service.py:95
Integration: Subaru (HACS) (documentation, issues)
First occurred: 4:28:12 PM (3 occurrences)
Last logged: 4:36:26 PM

Error executing service: <ServiceCall lock.lock (c:4fbb6cb2b0571fdff57819901fa6c6cc): entity_id=['lock.subaru_outback_door_locks']>
Error executing service: <ServiceCall lock.lock (c:15d00e272c1fdc645ce44808f5d1034b): entity_id=['lock.subaru_outback_door_locks']>
Error executing service: <ServiceCall lock.lock (c:b48fb47eeb2b1f244fbd38d28fe45e08): entity_id=['lock.subaru_outback_door_locks']>
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/core.py", line 1511, in catch_exceptions
    await coro_or_task
  File "/usr/src/homeassistant/homeassistant/core.py", line 1530, in _execute_service
    await handler.job.target(service_call)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_component.py", line 209, in handle_service
    await self.hass.helpers.service.entity_service_call(
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 663, in entity_service_call
    future.result()  # pop exception if have
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 896, in async_request_call
    await coro
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 700, in _handle_entity_call
    await result
  File "/config/custom_components/subaru/lock.py", line 67, in async_lock
    await async_call_remote_service(
  File "/config/custom_components/subaru/remote_service.py", line 95, in async_call_remote_service
    raise HomeAssistantError(f"Service {cmd} failed for {car_name}: {err_msg}")
homeassistant.exceptions.HomeAssistantError: Service lock failed for VictorC Outback: Remote service request completed but failed: NegativeAcknowledge:unknown
```

**Comment by stboch** [comment]
Sadly this error is a frequent one I got I have never been able to track it down... I sometimes get an error in the app too. usually I try again a few minutes later and it works, not sure what exactly the issue is but is seems like the Car polling to Subaru times out and fails with a generic error.

**Comment by G-Two** [comment]
@vmcosco the state of the door locks will always be unknown. The Subaru telematics system does not currently report door lock state, thus no state is passed by the underlying python package to HA. As for the `NegativeAcknowledge:unknown` error you received, it means the lock command made it all the way to your vehicle, but your vehicle reported that it is unable to lock for "unknown" reason. It could be something as simple as a door being open. Have you had any better luck recently?
